homeassistant/components/calendar/__init__.py

```diff
@@ -793,7 +793,9 @@ async def async_list_events_service(
         end = start + service_call.data[EVENT_DURATION]
     else:
         end = service_call.data[EVENT_END_DATETIME]
-    calendar_event_list = await calendar.async_get_events(calendar.hass, start, end)
+    calendar_event_list = await calendar.async_get_events(
+        calendar.hass, dt_util.as_local(start), dt_util.as_local(end)
```

Comment:
Do we need more tests for this?

Reply:
👍🏼 

(I commented out this change, added tests, verified they catch the issue (failure with floating vs tz-aware datetime comparison) then added the change back in.)

Reply:
Perfect!